ryot_compass/src/bevy_compass/drawing/brush/mod.rs

```diff
@@ -0,0 +1,73 @@
+use bevy::prelude::*;
+use ryot::prelude::{drawing::*, position::*, *};
+
+mod diamond;
+pub use diamond::DiamondBrush;
+
+mod geometric;
+pub use geometric::GeometricBrush;
+
+mod round;
+pub use round::RoundBrush;
+use ryot::bevy_ryot::AppearanceDescriptor;
+use ryot::position::TilePosition;
+
+mod square;
+pub use square::SquareBrush;
+
+mod systems;
+pub use systems::update_brush;
+
+pub trait BrushAction: Eq + PartialEq + Clone + Reflect + Send + Sync + 'static {
+    fn apply(&self, center: DrawingBundle) -> Vec<DrawingBundle>;
+    fn get_positions(&self, center: TilePosition) -> Vec<TilePosition> {
+        self.apply(DrawingBundle::new(
+            Layer::Ground,
+            center,
+            AppearanceDescriptor::default(),
+        ))
+        .into_iter()
+        .map(|bundle| bundle.tile_pos)
+        .collect()
+    }
+}
+
+#[derive(Component, Eq, Default, PartialEq, Reflect, Copy, Clone, Hash)]
+pub enum Brush {
+    #[default]
+    SingleTile,
+    Geometric(GeometricBrush),
+}
+
+impl Brush {
```

Comment:
I guess I see why you went with the class style now. but Why can't `SingleTile` be a geometric brush of size 1? It seems like an extra layer of abstraction that isn't helping with much here. But maybe you have a plan for this that I haven't gathered from the code.